containers/src/Data/Set/Internal.hs

```diff
@@ -953,6 +958,30 @@ mapMonotonic :: (a->b) -> Set a -> Set b
 mapMonotonic _ Tip = Tip
 mapMonotonic f (Bin sz x l r) = Bin sz (f x) (mapMonotonic f l) (mapMonotonic f r)
 
+
+-- | /O(n*log n)/.
+--
+-- Like 'traverse' from Data.Traversable. This is less generic, since 'Set'
+-- does not have a Traversable instance.
+traverse :: (Ord b, Applicative m) => (a -> m b) -> Set a -> m (Set b)
+traverse f s0 = fmap (GHCExts.fromListN (size s0)) . DT.traverse f . toList $ s0
```

Comment:
I know there are some subtle difference on whether to use `traverse f s0` vs `traverse f`, e.g. inlining. What's the verdict here? Do we want it half-pointfree or fully pointfree even?

Reply:
@massudaw any idea?